layers.go

```diff
@@ -524,27 +530,6 @@ func (s *store) newLayerStore(rundir string, layerdir string, driver drivers.Dri
 	return &rlstore, nil
 }
 
-func newROLayerStore(rundir string, layerdir string, driver drivers.Driver) (ROLayerStore, error) {
-	lockfile, err := GetROLockfile(filepath.Join(layerdir, "layers.lock"))
```

Comment:
`newLayerStore` as a replacement for this will always create a RW lockfile. Is that desirable? It changes semantics of `r.lockfile.RecursiveLock` == `r.RecursiveLock`, which seems worrisome.

OTOH https://github.com/containers/storage/blob/9f889830f10075aceb6b925a97927e77b837d0a7/pkg/lockfile/lockfile.go#L92-L100 does not support having both kinds of lock for a single path.